core/src/serve_repair.rs

```diff
@@ -445,9 +464,103 @@ impl ServeRepair {
         }
     }
 
+    fn decode_requests(
+        thread_pool: &ThreadPool,
+        reqs_v: Vec<PacketBatch>,
+        epoch_staked_nodes: &Option<Arc<HashMap<Pubkey, u64>>>,
+        whitelist: &RwLockReadGuard<HashSet<Pubkey>>,
+        my_id: &Pubkey,
+        socket_addr_space: &SocketAddrSpace,
+        stats: &mut ServeRepairStats,
+        cluster_type: ClusterType,
+    ) -> Vec<RepairRequestWithMeta> {
+        thread_pool
+            .install(|| {
+                reqs_v
+                    .par_iter()
+                    .with_min_len(VERIFY_MIN_PACKETS_PER_THREAD)
+                    .flatten()
+                    .filter_map(|packet| {
+                        let request: RepairProtocol = match packet.deserialize_slice(..) {
+                            Ok(request) => request,
+                            Err(_) => {
+                                stats.err_malformed.fetch_add(1, Ordering::Relaxed);
+                                return None;
+                            }
+                        };
+
+                        let from_addr = packet.meta().socket_addr();
+                        if !ContactInfo::is_valid_address(&from_addr, socket_addr_space) {
+                            stats.err_malformed.fetch_add(1, Ordering::Relaxed);
+                            return None;
+                        }
+
+                        match Self::verify_signed_packet(my_id, packet, &request) {
+                            Ok(()) => (),
+                            Err(e) => {
+                                match e {
+                                    Error::RepairVerify(RepairVerifyError::IdMismatch) => {
+                                        stats.err_id_mismatch.fetch_add(1, Ordering::Relaxed);
+                                    }
+                                    Error::RepairVerify(RepairVerifyError::Malformed) => {
+                                        stats.err_malformed.fetch_add(1, Ordering::Relaxed);
+                                    }
+                                    Error::RepairVerify(RepairVerifyError::SigVerify) => {
+                                        stats.err_sig_verify.fetch_add(1, Ordering::Relaxed);
+                                    }
+                                    Error::RepairVerify(RepairVerifyError::TimeSkew) => {
+                                        stats.err_time_skew.fetch_add(1, Ordering::Relaxed);
+                                    }
+                                    Error::RepairVerify(RepairVerifyError::Unsigned) => {
+                                        stats.err_unsigned.fetch_add(1, Ordering::Relaxed);
+                                    }
```

Comment:
why is it better to increment these err counters here as opposed to inside `verify_signed_packet` as before?